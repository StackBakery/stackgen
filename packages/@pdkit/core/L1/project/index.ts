export * from "./Project";
export * from "./Workspace";
