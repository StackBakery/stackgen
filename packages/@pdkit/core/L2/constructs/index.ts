export * from "./AppendFile";
export * from "./GitIgnore";
export * from "./License";
export * from "./Manifest";
export * from "./ManifestEntry";
