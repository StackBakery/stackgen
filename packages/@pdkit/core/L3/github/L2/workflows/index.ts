export * from "./BuildWorkflow";
export * from "./PullRequestLintWorkflow";
export * from "./ReleaseWorkflow";
