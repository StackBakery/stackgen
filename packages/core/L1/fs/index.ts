export * from "./File";
export * from "./JsonFile";
export * from "./YamlFile";
export * from "./TemplateFile";
