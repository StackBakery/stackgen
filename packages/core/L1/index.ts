export * from "./traits";
export * from "./fs";
export * from "./project";
export * from "./scripts";
export * from "./synthesizers";
