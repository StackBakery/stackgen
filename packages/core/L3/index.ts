export * from "./cloud";
export * from "./github";
export * from "./nodejs";
