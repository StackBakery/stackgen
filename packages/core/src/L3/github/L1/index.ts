export * from "./GithubJob";
export * from "./GithubStep";
export * from "./GithubWorkflow";
