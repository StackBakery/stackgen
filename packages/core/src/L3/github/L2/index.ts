export * from "./jobs";
export * from "./steps";
export * from "./workflows";
export * from "./GithubSupport";
