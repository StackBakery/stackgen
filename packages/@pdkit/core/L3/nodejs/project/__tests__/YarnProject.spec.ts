import { PostInstallShellScript } from "../../../../L1";
import { Manifest } from "../../../../L2";
import { YarnProject, YarnWorkspace } from "../../../../L3";

describe("YarnProject", () => {
  it("creates a generic YarnProject", () => {
    const workspace = new YarnWorkspace();

    const project = new YarnProject(workspace, "MyYarnProject", {
      description: "MyDescription",
      repository: {
        type: "git",
        url: "http://localhost",
      },
      private: true,
      resolutions: { a: "b" },
    });

    workspace.synth();

    const manifest = Manifest.of(project);
    const fields = manifest.fields;

    expect(fields.name).toBe("my-yarn-project");
    expect(fields.description).toBe("MyDescription");
    expect(fields.private).toBe(true);
    expect(fields.main).toBe("build/index.js");
    expect(fields.files).toStrictEqual(["build/*.js", "build/**/*.js"]);
    expect(fields.devDependencies).toStrictEqual({ "@pdkit/cli": "*", "@pdkit/core": "*" });
    expect(fields.resolutions).toStrictEqual({ a: "b" });
    expect(fields.repository).toStrictEqual({
      type: "git",
      url: "http://localhost",
    });
    expect(fields.jest).toBeUndefined();
    expect(fields.eslintConfig).toBeUndefined();
    expect(fields.publishConfig).toBeUndefined();
    expect(fields.scripts).toBeUndefined();
    expect(fields.workspaces).toBeUndefined();
    expect(fields.a).toBeUndefined();
    expect(manifest.fields).toMatchSnapshot();
  });

  it("creates a generic YarnProject with Yalc", () => {
    const workspace = new YarnWorkspace();

    const project = new YarnProject(workspace, "MyYarnProject", {
      description: "MyDescription",
      repository: {
        type: "git",
        url: "http://localhost",
      },
      private: true,
      resolutions: { a: "b" },
      yalc: true,
    });

    workspace.synth();

    const manifest = Manifest.of(project);
    const fields = manifest.fields;

    expect(fields.name).toBe("my-yarn-project");
    expect(fields.description).toBe("MyDescription");
    expect(fields.private).toBe(true);
    expect(fields.main).toBe("build/index.js");
    expect(fields.files).toStrictEqual(["build/*.js", "build/**/*.js"]);
    expect(fields.devDependencies).toStrictEqual({ "@pdkit/cli": "*", "@pdkit/core": "*" });
    expect(fields.resolutions).toStrictEqual({ a: "b" });
    expect(fields.repository).toStrictEqual({
      type: "git",
      url: "http://localhost",
    });
    expect(fields.jest).toBeUndefined();
    expect(fields.eslintConfig).toBeUndefined();
    expect(fields.publishConfig).toBeUndefined();
    expect(fields.scripts).toStrictEqual({ yalc: "npx yalc publish" });
    expect(fields.workspaces).toBeUndefined();
    expect(fields.a).toBeUndefined();
    expect(manifest.fields).toMatchSnapshot();

    const installCommand = project.node.tryFindChild("InstallCommand") as PostInstallShellScript;

    expect(installCommand).not.toBeUndefined();
    expect(installCommand.command).toStrictEqual(["yarn"]);
  });
});
