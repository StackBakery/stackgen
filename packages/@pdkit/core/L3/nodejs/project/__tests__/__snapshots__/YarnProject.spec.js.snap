// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YarnProject creates a generic YarnProject 1`] = `
Object {
  "bin": undefined,
  "bugs": undefined,
  "description": "MyDescription",
  "devDependencies": Object {
    "@pdkit/cli": "*",
    "@pdkit/core": "*",
  },
  "files": Array [
    "build/*.js",
    "build/**/*.js",
  ],
  "homepath": undefined,
  "keywords": undefined,
  "main": "build/index.js",
  "man": undefined,
  "name": "my-yarn-project",
  "private": true,
  "repository": Object {
    "type": "git",
    "url": "http://localhost",
  },
  "resolutions": Object {
    "a": "b",
  },
  "scripts": undefined,
  "types": undefined,
  "version": "3.0.11",
}
`;

exports[`YarnProject creates a generic YarnProject with Yalc 1`] = `
Object {
  "bin": undefined,
  "bugs": undefined,
  "description": "MyDescription",
  "devDependencies": Object {
    "@pdkit/cli": "*",
    "@pdkit/core": "*",
  },
  "files": Array [
    "build/*.js",
    "build/**/*.js",
  ],
  "homepath": undefined,
  "keywords": undefined,
  "main": "build/index.js",
  "man": undefined,
  "name": "my-yarn-project",
  "private": true,
  "repository": Object {
    "type": "git",
    "url": "http://localhost",
  },
  "resolutions": Object {
    "a": "b",
  },
  "scripts": Object {
    "yalc": "npx yalc publish",
  },
  "types": undefined,
  "version": "3.0.11",
}
`;
